staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -233,6 +239,18 @@ type ClaimValidationRule struct {
 	// If claim is set and requiredValue is not set, the claim must be present with a value set to the empty string.
 	// +optional
 	RequiredValue string `json:"requiredValue"`
+
+	// expression is a logical expression that is written in CEL https://github.com/google/cel-go.
+	// Must return true for the validation to pass.
+	// Mutually exclusive with claim and requiredValue.
+	// +optional
+	Expression string `json:"expression"`
+	// message customizes the returned error message when expression returns false.
+	// Mutually exclusive with claim and requiredValue.
+	// Note that messageExpression is explicitly not supported to avoid
+	// misconfigured expressions from leaking JWT payload contents.
+	// +optional
+	Message string `json:"message,omitempty"`
```

Comment:
Do we need CEL expression support for when producing messages? We've ended up needing this is most of the other places where we offer CEL expressions.